packages/features/bookings/components/VerifyCodeDialog.tsx

```diff
@@ -25,6 +25,7 @@ export const VerifyCodeDialog = ({
   const [isLoading, setIsLoading] = useState(false);
   const [error, setError] = useState("");
   const [value, setValue] = useState("");
+  const [hasVerified, setHasVerified] = useState(false);
```

Comment:
the reason for multiple bookings was that verifyCode was being called multipe times, so create a flag and setting it to true once the function has been visited